File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;
        vector<vector<int>> freq(nums.size() + 1);

        for(int n : nums){
            count[n] = 1 + count[n];
        }
        for(const auto& kv : count){
            freq[kv.second].push_back(kv.first);
        }

        vector<int> result;
        for(int i = freq.size() - 1; i > 0; --i){
            for(int j : freq[i]){
                result.push_back(j);
                if(result.size() == k){
                    return result;
                }
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seq = set(nums)
        count, longest = 0, 0
        for i in nums:
            if (i - 1) not in seq:
                count = 0
                while (i + count) in seq:
                    count += 1
                if count > longest:
                    longest = count
        return longest

